---
title: "Coevolution of the haplosclerid holo-microbiome in the Red Sea"
author: "Sergio Vargas et al."
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Introduction

Sponge microbiomes are diverse and likely coevolve with their hosts. Here we analyze the microbiome of a large sample of haplosclerid sponges from the Red Sea. We use whole-genome clade specific markers to infer a robust phylogeny of these sponges and understand the evolution of the haplosclerid holo-microbiome.

```{r prep}

if(!requireNamespace("dada2", quietly = TRUE)){
  print("dada2 not installed, installing now...")
  if(!requireNamespace("BiocManager", quietly = TRUE)) {
    install.packages("BiocManager")
    install_exit_status <- BiocManager::install("dada2", lib="/home/sergio.vargas/R/x86_64-suse-linux-gnu-library/4.3")    
  }else{
    install_exit_status <- BiocManager::install("dada2", lib="/home/sergio.vargas/R/x86_64-suse-linux-gnu-library/4.3")  
  }
}else{
  print("dada2 there, loading...")
  library("dada2")
}

```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
