---
title: "Coevolution of the haplosclerid holo-microbiome in the Red Sea"
author: "Sergio Vargas et al."
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Introduction

Sponge microbiomes are diverse and likely coevolve with their hosts. Here we analyze the microbiome of a large sample of haplosclerid sponges from the Red Sea. We use whole-genome clade specific markers to infer a robust phylogeny of these sponges and understand the evolution of the haplosclerid holo-microbiome.

The following code is based on the *dada2* tutorial available at [https://benjjneb.github.io/dada2/tutorial.html](https://benjjneb.github.io/dada2/tutorial.html)


```{r prep}

if(!requireNamespace("dada2", quietly = TRUE)){
  print("dada2 not installed, installing now...")
  if(!requireNamespace("BiocManager", quietly = TRUE)) {
    install.packages("BiocManager")
    install_exit_status <- BiocManager::install("dada2", lib="/home/sergio.vargas/R/x86_64-suse-linux-gnu-library/4.3")    
  }else{
    install_exit_status <- BiocManager::install("dada2", lib="/home/sergio.vargas/R/x86_64-suse-linux-gnu-library/4.3")  
  }
}else{
  print("dada2 there, loading...")
  library("dada2")
}

```

## Load the clean reads

```{r load reads}

path <- "/mnt/scratch/Red_Sea_Microbiomes/Reads"
#list.files(path)

# Forward and reverse fastq filenames have format: SAMPLENAME_R1_001.fastq and SAMPLENAME_R2_001.fastq
fnFs <- sort(list.files(path, pattern="_R1_001.clean.fastq.gz", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="_R2_001.clean.fastq.gz", full.names = TRUE))

# Extract sample names, assuming filenames have format: SAMPLENAME_XXX.fastq
sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)

plotQualityProfile(fnFs[1:2])
plotQualityProfile(fnRs[1:2])

```


##  Place filtered files in filtered/ subdirectory
```{r filter reads}

filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_dada2.clean.fastq.gz"))
filtRs <- file.path(path, "filtered", paste0(sample.names, "_R_dada2.clean.fastq.gz"))
names(filtFs) <- sample.names
names(filtRs) <- sample.names

filter_output <- filterAndTrim(fnFs, filtFs, fnRs, filtRs,
              maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE,
              compress=TRUE, multithread=TRUE)

```

```{r learn errors}

errF <- learnErrors(filtFs, multithread=TRUE)
errR <- learnErrors(filtRs, multithread=TRUE)

plotErrors(errF, nominalQ=TRUE)

```

```{r Sample inference}

dadaFs <- dada(filtFs, err=errF, multithread=TRUE)
dadaRs <- dada(filtRs, err=errR, multithread=TRUE)

```

```{r merge reads}

merged_pairs <- mergePairs(dadaFs, filtFs, dadaRs, filtRs, verbose=TRUE)

```


```{r Get sequence table}

seqtab <- makeSequenceTable(merged_pairs)
seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=TRUE, verbose=TRUE)
sum(seqtab.nochim)/sum(seqtab)

```
```{r track reads}

getN <- function(x) sum(getUniques(x))
track <- cbind(filter_output, sapply(dadaFs, getN), sapply(dadaRs, getN), sapply(merged_pairs, getN), rowSums(seqtab.nochim))
# If processing a single sample, remove the sapply calls: e.g. replace sapply(dadaFs, getN) with getN(dadaFs)
colnames(track) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim")
rownames(track) <- sample.names

```
```{r Assign taxonomy}

red_sea_taxa <- assignTaxonomy(seqtab.nochim, "/mnt/scratch/Red_Sea_Microbiomes/SilvaDB/v138/silva_nr99_v138.1_train_set.fa.gz", multithread=TRUE)

red_sea_taxa <- addSpecies(red_sea_taxa, "/mnt/scratch/Red_Sea_Microbiomes/SilvaDB/v138/silva_species_assignment_v138.1.fa.gz")

```

