---
title: "Coevolution of the haplosclerid holo-microbiome in the Red Sea"
author: "Sergio Vargas et al."
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Introduction

Sponge microbiomes are diverse and likely coevolve with their hosts. Here we analyze the microbiome of a large sample of haplosclerid sponges from the Red Sea. We use whole-genome clade specific markers to infer a robust phylogeny of these sponges and understand the evolution of the haplosclerid holo-microbiome.

The following code is based on the *dada2* tutorial available at [https://benjjneb.github.io/dada2/tutorial_1_8.html](https://benjjneb.github.io/dada2/tutorial_1_8.html)

and this other tutorial I found:

[https://astrobiomike.github.io/amplicon/dada2_workflow_ex](https://astrobiomike.github.io/amplicon/dada2_workflow_ex)


```{r prep}

if(!requireNamespace("dada2", quietly = TRUE)){
  print("dada2 not installed, installing now...")
  if(!requireNamespace("BiocManager", quietly = TRUE)) {
    install.packages("BiocManager")
    install_exit_status <- BiocManager::install(c("dada2", "phyloseq"), lib="/home/sergio.vargas/R/x86_64-suse-linux-gnu-library/4.3")
  }else{
    install_exit_status <- BiocManager::install(c("dada2","phyloseq"), lib="/home/sergio.vargas/R/x86_64-suse-linux-gnu-library/4.3")  
  }
}else{
  print("dada2 there, loading...")
  library(dada2)
  library(phyloseq)
  library(ggplot2)
}

```

## Load the clean reads

Samples "GW3237", "GW3325", "GW3404", "GW3416", "GW3494", "GW3580", "GW4166", "GW4251", "GW5875", "GW6059", "GW6104", "GW6140", "GW6158", and "GW6196" were discarded due to low read counts (<5000).

Samples "GW3524", "GW4232", "GW5899", and "GW6180" were discarded due to missing UCE data.



```{r load reads}

path <- "/mnt/scratch/Red_Sea_Prokaryotic_Microbiomes/Reads/"
#list.files(path)

# Forward and reverse fastq filenames have format: SAMPLENAME_R1_001.fastq and SAMPLENAME_R2_001.fastq
fnFs <- sort(list.files(path, pattern="_R1_001.clean.fastq.gz", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="_R2_001.clean.fastq.gz", full.names = TRUE))

# Extract sample names, assuming filenames have format: SAMPLENAME_XXX.fastq
sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)

plotQualityProfile(fnFs[1:2])
plotQualityProfile(fnRs[1:2])

```


##  Place filtered files in filtered/ subdirectory
```{r filter reads}

filtFs <- file.path(path, "Dada2_Filtered", paste0(sample.names, "_F_dada2.clean.fastq.gz"))
filtRs <- file.path(path, "Dada2_Filtered", paste0(sample.names, "_R_dada2.clean.fastq.gz"))
names(filtFs) <- sample.names
names(filtRs) <- sample.names

filter_output <- filterAndTrim(fnFs, filtFs, fnRs, filtRs,
              maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE,
              compress=TRUE, multithread=TRUE)

```

```{r learn errors}

errF <- learnErrors(filtFs, multithread=TRUE)
errR <- learnErrors(filtRs, multithread=TRUE)

plotErrors(errF, nominalQ=TRUE)

```

```{r Sample inference}

dadaFs <- dada(filtFs, err=errF, multithread=TRUE)
dadaRs <- dada(filtRs, err=errR, multithread=TRUE)

```

```{r merge reads}

merged_pairs <- mergePairs(dadaFs, filtFs, dadaRs, filtRs, verbose=TRUE)

```


```{r Get sequence table}

seqtab <- makeSequenceTable(merged_pairs)
seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=TRUE, verbose=TRUE)
sum(seqtab.nochim)/sum(seqtab)

```
```{r track reads}

getN <- function(x) sum(getUniques(x))
track <- cbind(filter_output, sapply(dadaFs, getN), sapply(dadaRs, getN), sapply(merged_pairs, getN), rowSums(seqtab.nochim), rowSums(seqtab.nochim)/filter_output[,2])
# If processing a single sample, remove the sapply calls: e.g. replace sapply(dadaFs, getN) with getN(dadaFs)
colnames(track) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim", "surviving reads")
rownames(track) <- sample.names
#rownames(track[track[,1]<5000,])

```

```{r Assign taxonomy}

red_sea_taxa <- assignTaxonomy(seqtab.nochim,"/mnt/scratch/Red_Sea_Prokaryotic_Microbiomes/SilvaDB/v138/silva_nr99_v138.1_train_set.fa.gz", multithread=TRUE)

red_sea_taxa <- addSpecies(red_sea_taxa, "/mnt/scratch/Red_Sea_Prokaryotic_Microbiomes/SilvaDB/v138/silva_species_assignment_v138.1.fa.gz")

```

```{r Inspect taxonomy table}

red_sea_taxa.print <- red_sea_taxa # Removing sequence rownames for display only
rownames(red_sea_taxa.print) <- NULL
head(red_sea_taxa.print)


```

```{r extract ASV table, etc}

# giving our seq headers more manageable names (ASV_1, ASV_2...)
asv_seqs <- colnames(seqtab.nochim)
asv_headers <- vector(dim(seqtab.nochim)[2], mode="character")

for (i in 1:dim(seqtab.nochim)[2]) {
    asv_headers[i] <- paste(">ASV", i, sep="_")
}

# making and writing out a fasta of our final ASV seqs:
asv_fasta <- c(rbind(asv_headers, asv_seqs))
write(asv_fasta, "./Outfiles/ASVs.fa")

# count table:
asv_tab <- t(seqtab.nochim)
row.names(asv_tab) <- sub(">", "", asv_headers)
write.table(asv_tab, "./Outfiles/ASVs_counts.tsv", sep="\t", quote=F, col.names=NA)

#taxonomy table:
rownames(red_sea_taxa.print) <- asv_headers
write.table(red_sea_taxa.print, "./Outfiles/ASVs_taxonomy.tsv", sep = "\t", quote=F, col.names=NA)

```


<!-- ```{r Prep phyloseq object} -->

<!-- theme_set(theme_bw()) -->
<!-- samples.out <- rownames(seqtab.nochim) -->
<!-- #here load a df with sample to sample info for figure annotations, I guess -->
<!-- #samdf <- data.frame(Subject=subject, Gender=gender, Day=day) -->
<!-- #samdf$When <- "Early" -->
<!-- #samdf$When[samdf$Day>100] <- "Late" -->
<!-- #rownames(samdf) <- samples.out -->


<!-- ps <- phyloseq(otu_table(seqtab.nochim, taxa_are_rows=FALSE),  -->
<!--                #sample_data(samdf),# add sample data here. -->
<!--                tax_table(red_sea_taxa)) -->

<!-- plot_richness(ps, measures=c("Shannon", "Simpson"))#, x="Day", , color="When") -->


<!-- ps.prop <- transform_sample_counts(ps, function(otu) otu/sum(otu)) -->
<!-- ord.nmds.bray <- ordinate(ps.prop, method="NMDS", distance="bray", trymax=100) -->

<!-- plot_ordination(ps, ord.nmds.bray, title="Bray NMDS") -->

<!-- ``` -->


